### DISCUSSION TITLE
URL Submission Process and File Locations

### DISCUSSION THREAD

**Discussion by dhoule01** [CMSgov/price-transparency-guide#128] [General]
On the main github page under "Public Discoverability" it says "These machine-readable files must be made available to the public without restrictions that would impede the re-use of that information. The location of the these URLs must be provided over HTTPS to ensure the integrity of the data."

1)  How will we be submitting these URL locations to CMS?  Through the already existing JSON Machine Readable index URL submission for plans, drugs, provider files?  Or through some other URL submission mechanism specific to price transparency?

2)  Will it be allowable to create separate files for individual MA and group MA and serve them from different hosts, for example:

https://vendorONE.org/files/2020-01-05_cms_medicare_IND_in-network-rates.json
https://vendorONE.org/files/2020-01-05_cms_medicare_IND_allowed-amounts.json
https://vendorONE.org/files/2020-01-05_cms_medicare_IND_prescription-drugs.json
and
https://vendorTWO.org/files/2020-01-05_cms_medicare_GRP_in-network-rates.json
https://vendorTWO.org/files/2020-01-05_cms_medicare_GRP_allowed-amounts.json
https://vendorTWO.org/files/2020-01-05_cms_medicare_GRP_prescription-drugs.json

and then submit all 6 paths in the CMS URL submission process?

**Comment by ghost** [comment]
@dhoule01 For your question #1 you might be reading too much into the text you quoted. I do not believe we are required to provide CMS a listing of the URL locations. We are to provide a public posting of our files accessible via URL over HTTPS.

I do not understand your reference to "MA" on your second questions so I can not provide any comment.

**Reply by dhoule01** [reply]
Yes, it says to provide a public posting of the URLs... but how will anyone find them to use them if their location isn't published somewhere?

With the existing machine readable JSON files we submit a single URL to CMS along with out payer id, that URL points to our index file, which contains URLs to our plans, drugs, providers JSON files.  CMS keeps track of the submission of our index file URL to a) show our payer id is complying, b) come and get the index daily to c) download our JSON files, d) and make that index URL available publicly so various other parties can pick them up as well.

My assumption is that we'd be using the existing machine readable index submission process or something similar.  If there's no mechanism like this for price transparency, how will the public know where to pick them up?  What's the point of putting these price transparency files out there for the public if the public can't find them?  And if you don't submit their URL locations to CMS how will CMS know you've complied and put them out there?

And MA is our company's internal abbreviation for Medicare

**Comment by Devyandu** [comment]
The current regulation does not require us to submit the URLs to CMS.

They just need to be on the payer's site, available via a public URL.

**Reply by dhoule01** [reply]
I agree, but doesn't common sense dictate they'll need to add this?  What's the point in doing all this work and putting all this data out there if no one can find it?

**Comment by Devyandu** [comment]
You are right.
Perhaps we will get future guidance on discoverability/cataloging aspects
of the machine readable files, but it is not there today.

**Reply by ghost** [reply]
@Devyandu That is going to be our approach for posting the files.

**Reply by dhoule01** [reply]
Thank you, I read the FR more closely and see now you're right, but with caveats...


https://www.govinfo.gov/content/pkg/FR-2020-11-12/pdf/2020-24591.pdf

Page 85, right column, under section 3, about halfway down the page:

"The Departments also considered requiring plans and issuers to submit the internet addresses for the machine-readable files to CMS, and having CMS make the information available to the public."

Subsequent discussion, and then...

Page 86, left column, at the bottom:

"At this time, the Departments are of the view that reporting of the links to the file locations is not necessary to achieve the goals of the final rules. However, the Departments note that nothing in the final rules prevents a Federal or state regulatory body, such as a state Department of Insurance (DOI), from collecting this information from issuers subject to their jurisdiction."

This tells me there will need to be a way to announce or publish the locations of these links, otherwise there would be no way for anyone to use them.  So there's no requirement for link reporting at this time, however I expect there will be, and it will come quickly.


My second question still stands... is a payer required to provide:

all in network rates in one in-network-rates.json file
all allowed amounts in one allowed-amounts.json file
all prescription drugs in one prescription-drugs.json file

... or... can a payer provide

some in network rates in one in-network-rates.json file and some in another file
Same for allowed amounts
Same for prescription drugs
???

**Reply by ghost** [reply]
@dhoule01 Based on the Github CMS schema, the single JSON file is based on a plan where the root JSON data elements list the Plan Name, Plan ID Type, Plan ID and Plan Market Type. This is where the Plan is associated with a single HIOS or EIN based on the Plan ID field. If you have plans that have different HIOS/EIN values, that would require them to be reported in separate files for each of the 3 file types per the naming convention. If you see in some of the other discussion questions, insurers have found where a plan name is not unique as there can be multiple HIOS that relate to a single plan name asking for CMS guidance for that situation.

There is the proposal #93 where we recognize that negotiated rates are duplicated for multiple plans and this would be a means to eliminate the duplication and still provide the related Plan data elements and to support having multiple HIOS/EINs within a single JSON file with the associated negotiated rates.

**Comment by Devyandu** [comment]
There is an existing discussion on the number of files here -
https://github.com/CMSgov/price-transparency-guide/discussions/93

The naming convention
https://github.com/CMSgov/price-transparency-guide#file-naming-convention
implies that there are 3 files per plan (in-network. allowed-amount and
prescription).

PS : This is my interpretation, not a CMS certified response :-)
